### PR TITLE
MDEV-11866: (10.2) ANALYZE FORMAT=JSON not predicatable for r_total_time_ms/r_buf…

### PR DESCRIPTION
**Pull request by grooverdan**
…fer_size

[0-9]*[.]?[0-9]* wasn't a sufficient regex to cover the
%lg used in Json_writer::add_double. Exponent formats
where missed.

Here we normalize all the replace_regex expressions for
ANALYZE FORMAT=JSON into one include file.

I submit this under the MCA.

### CODE REVIEW

File: mysql-test/t/cte_recursive.test [hunk 1]
```diff
@@ -1304,6 +1304,7 @@ drop table folks;
 create table t1(a int);
 insert into t1 values (0),(1),(2),(3),(4),(5),(6),(7),(8),(9);
  
+--source include/analyze-format.inc
```
Comment (spetrunia) [inline mysql-test/t/cte_recursive.test]: I'm wondering why is this necessary here?  The statement is `EXPLAIN`, not `ANALYZE`.  EXPLAIN does not produce any execution-time members (those that start with `r_`, like r_total_time_ms or r_buffer_size).   
That fact that this patch doesn't change the r/cte_recursive.result file proves this point

File: mysql-test/include/analyze-format.inc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# The time on ANALYSE FORMAT=JSON is rather variable
+
+--replace_regex /("(r_total_time_ms|r_buffer_size)": )[^, \n]*/\1"REPLACED"/
```
Comment (spetrunia) [inline mysql-test/include/analyze-format.inc]: I have only one issue: .result files are used in discussions about what other `r_` parameters should be added, so it would be nicer if the field names remained intact.   Which should be easy to do as `replace_regex` supports multiple replaces, something like:
`
+--replace_regex /("(r_total_time_ms)": )[^, \n]*/\1"REPLACED"/   /("(r_buffer_size)": )[^, \n]*/\1"REPLACED"/ 
`

File: mysql-test/t/cte_recursive.test (hunk 1 above)
Comment (grooverdan) [inline mysql-test/t/cte_recursive.test]: doh. of course. I'll look closer for the right line.

File: mysql-test/include/analyze-format.inc (hunk 2 above)
Comment (grooverdan) [inline mysql-test/include/analyze-format.inc]: The field names are intact as indicated by the result files. The outer brackets form a capture and \1 places the capture in the result and only replaces the value of the parameter.

File: mysql-test/t/cte_recursive.test (hunk 1 above)
Comment (grooverdan) [inline mysql-test/t/cte_recursive.test]: Ok, removed this part of the patch. Travis reported the error as per the JIRA bug but I can't see how it occurred. Will add something later if I find it.

**Comment by grooverdan** [comment]
As I can't remove the cte commit I'm closing this and letting the 10.1 pull request #294 continue. There is a 10.2 differences in the analyze_format_json.test which need to be re-recorded on merge.

File: mysql-test/t/cte_recursive.test (hunk 1 above)
Comment (elenst) [inline mysql-test/t/cte_recursive.test]: The failure in cte_recursive happens, not here, but in cte_recursive.test:1514 (which corresponds to cte_recursive.result:2367).

Comment (grooverdan) [inline mysql-test/t/cte_recursive.test]: ah thanks. I wasn't up to date to abfcdb8f where the failure occurred. Thanks for tracing it down.
